randovania/games/prime_hunters/exporter/patch_data_factory.py

```diff
@@ -52,16 +61,53 @@ def force_field_index_for_requirement(game: GameDescription, requirement: Layout
     "Octolith8": 7,
 }
 
+_ITEM_TO_ITEM_TYPE = {
+    "Energy Tank": 4,
+    "Volt Driver": 5,
+    "Missile": 6,
+    "Battlehammer": 7,
+    "Imperialist": 8,
+    "Judicator": 9,
+    "Magmaul": 10,
+    "Shock Coil": 11,
+}
+
 
 class HuntersPatchDataFactory(PatchDataFactory[HuntersConfiguration, HuntersCosmeticPatches]):
     def game_enum(self) -> RandovaniaGame:
         return RandovaniaGame.METROID_PRIME_HUNTERS
 
-    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str]:
-        result = {
-            # TODO: Take starting weapons into account
-            "weapons_string": "00000101",
-        }
+    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str | int]:
+        result: dict[str, str | int] = {}
+        starting_items = {}
+
+        for resource, quantity in resources.as_resource_gain():
+            try:
+                starting_items[item_type_for_item_resource(resource)] = quantity
+            except KeyError:
+                print(f"Skipping {resource} for starting inventory: no item id")
+                continue
+
+        weapons = [
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Shock Coil"], 0),
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Magmaul"], 0),
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Judicator"], 0),
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Imperialist"], 0),
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Battlehammer"], 0),
+            1,  # Missiles
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Volt Driver"], 0),
+            1,  # Power Beam
+        ]
+        starting_weapons = "".join(map(str, weapons))
+        result["weapons"] = starting_weapons
+
+        # The value of missiles ends up being starting + 1, so subtract 1
+        starting_missiles = starting_items.get(_ITEM_TO_ITEM_TYPE["Missile"], 1) - 1
```

Comment:
this is pretty icky. shouldn't this sorta case be handled in the patcher?

Reply:
I think i wrote it this way because when I was getting the value of missiles from the preset, it was always +1 the value that was set. I just want the amount of missiles given as configured by the launcher